### PR TITLE
Remove Go 1.6 code from moby

### PR DESCRIPTION
**Pull request by boaz0**
As suggested [here](https://github.com/moby/moby/issues/32120#issuecomment-289492614), once Go 1.6 will be deprecated, there is no need to save code with the `build go1.6` tag.

**- What I did**

Remove [api/server/httputils/httputils_write_json_go16.go](https://github.com/moby/moby/blob/4f259698b07653e9e5220e097df79862f9e54b74/api/server/httputils/httputils_write_json_go16.go) and [pkg/tlsconfig/tlsconfig_clone_go16.go](https://github.com/moby/moby/blob/4f259698b07653e9e5220e097df79862f9e54b74/pkg/tlsconfig/tlsconfig_clone_go16.go).

**- How I did it**

Literally deleted the files.

**- How to verify it**

Run tests (unit and integration)

### CODE REVIEW

**Comment by ripcurld0** [comment]
@runcom can you please review?

File: api/server/httputils/httputils_write_json_go16.go [hunk 1]
```diff
@@ -1,16 +0,0 @@
-// +build go1.6,!go1.7
```
Comment (runcom) [inline api/server/httputils/httputils_write_json_go16.go]: you could remove the build tag from here I guess https://github.com/moby/moby/blob/4f259698b07653e9e5220e097df79862f9e54b74/api/server/httputils/httputils_write_json.go#L1

**Comment by runcom** [comment]
LGTM (minus a nit)

File: api/server/httputils/httputils_write_json_go16.go (hunk 1 above)
Comment (ripcurld0) [inline api/server/httputils/httputils_write_json_go16.go]: Done! Thanks 😸

**Review by vdemeester — APPROVED** [review]
LGTM 🐯

**Comment by vieux** [comment]
LGTM

**Pull request merged**